src/components/organisms/Footer/data.ts

```diff
@@ -0,0 +1,53 @@
+import { menuItemData } from '@/components/molecules/data';
+import { fetchAPI } from '@/lib';
+import { gql } from '@/utils';
+
+import block from './block.json';
+
+// Export block slug for identification
+export const slug = block.slug;
+
+export const formatter = (data: GraphQLFooterFields): FooterData => ({
+	menus: {
+		footer: { items: menuItemData.formatter(data.footer?.nodes ?? []) },
+		legal: { items: menuItemData.formatter(data.legal?.nodes ?? []) },
+		social: { items: menuItemData.formatter(data.social?.nodes ?? []) },
+	},
+	siteTitle: data.generalSettings?.title ?? 'Superstack',
+	isHome: data.node?.isFrontPage ?? false,
+});
+
+export const getData = async (uri: string): Promise<FooterData> => {
+	const query = gql`
+		query footerQuery($uri: String!) {
+			footer: menuItems(where: { location: FOOTER }, first: 9999) {
+				nodes {
+					...menuItemFragment
+				}
+			}
+			legal: menuItems(where: { location: LEGAL }, first: 9999) {
+				nodes {
+					...menuItemFragment
+				}
+			}
+			social: menuItems(where: { location: SOCIAL }, first: 9999) {
+				nodes {
+					...menuItemFragment
+				}
+			}
+			generalSettings {
+				title
+			}
+			node: nodeByUri(uri: $uri) {
+				... on Page {
+					isFrontPage
+				}
+			}
```

Comment:
@kuuak Maybe I'm completely wrong, but I'm just wondering, does passing the `$uri` as a variable of `getData` means it must re-fetch the data of the footer/header for each page ? And also if we go from one page to an other, it means it will re-render the whole footer/header as its data changed ?
Maybe it's not an issue at all, but I'm not sure anymore 🙈

Reply:
You're right, the nodeByUri will be re-queried for the Header & Footer additionaly to the getNodeByUri query.

However the should be way faster as it only returns one field. Also i think that nextjs now with the App router, avoids to re-render the layout if it's not needed (to be confirmed)